src/modes/run.ts

```diff
@@ -1,5 +1,340 @@
+
+      // If we are in sync, let it be known
+      if (currentBlockNumber === this.registry.lastProcessedBlock) {
+        this.inSync = true;
+      } else {
+        // Otherwise, we need to keep processing blocks
+        this.inSync = false;
+        fromBlock = this.registry.lastProcessedBlock + 1;
+        plan = {};
+      }
+    } while (!this.inSync);
+
+    _(oneShot ? "Chain watcher is in sync" : "Chain watcher is warmed up");
+    _(`Last processed block: ${this.registry.lastProcessedBlock}`);
+
+    // Notifying that the chain watcher is in sync
+    this.inSync = true;
+
+    // If one-shot, return
+    if (oneShot) {
+      return;
+    }
+
+    // Otherwise, run the block watcher
+    return await this.runBlockWatcher();
+  }
+
+  /**
+   * Run the block watcher for the chain. As new blocks come in:
+   * 1. Check if there are any `ConditionalOrderCreated` events, and index these.
+   * 2. Check if any orders want to create discrete orders.
+   */
+  public async runBlockWatcher() {
+    const { provider, chainId } = this.chainContext;
+    const _ = (s: string) =>
+      console.log(`[runBlockWatcher:chainId:${chainId}] ${s}`);
+    // Watch for new blocks
+    _("Subscribe to new blocks");
+    provider.on("block", async (blockNumber: number) => {
+      try {
+        _(`New block ${blockNumber}`);
+
+        const events = await pollContractForEvents(
+          blockNumber,
+          blockNumber,
+          this.chainContext
+        );
+
+        try {
+          await processBlock(this, Number(blockNumber), events);
+        } catch {
+          console.error(
+            `[runBlockWatcher:chainId:${chainId}] Error processing block ${blockNumber}`
+          );
+        }
+        _(`Block ${blockNumber} has been processed.`);
+      } catch (error) {
+        console.error(
+          `[runBlockWatcher:chainId:${chainId}] Error in processBlock`,
+          error
+        );
+      }
+    });
+  }
+}
+
+async function processBlock(
+  chain: ChainWatcher,
+  blockNumber: number,
+  events: ConditionalOrderCreatedEvent[],
+  blockNumberOverride?: number,
+  blockTimestampOverride?: number
+) {
+  const { provider } = chain.chainContext;
+  const block = await provider.getBlock(blockNumber);
+
+  // Transaction watcher for adding new contracts
+  let hasErrors = false;
+  for (const event of events) {
+    const receipt = await provider.getTransactionReceipt(event.transactionHash);
+    if (receipt) {
+      // run action
+      console.log(
+        `[processBlock] Run "addContract" action for TX ${event.transactionHash}`
+      );
+      const result = await addContract(chain, event)
+        .then(() => true)
+        .catch((e) => {
+          hasErrors = true;
+          console.error(
+            `[run_local] Error running "addContract" action for TX:`,
+            e
+          );
+          return false;
+        });
+      console.log(
+        `[run_local] Result of "addContract" action for TX ${
```

Comment:
There 2 logs do not use the same prefix as the other log msgs.
Is this intentional?

Reply:
Tracked in #60